tools/Generator/Generator.cs

```diff
@@ -1369,7 +1369,11 @@ bool WriteCommand (XElement commandElement, bool prependNewLine, bool isForHandl
 				WriteCommandParameters (commandElement, null, dataParam, null, isForHandle && !isExtension, true, paramsDict, isExtension);
 				WriteLine (");");
 				CommandHandleResult (hasResult);
-				WriteLine ();
+                IndentWriteLine("if ({0} <= 0)", intParam.csName);
+                IndentLevel++;
+                IndentWriteLine("return new {0}[0];", dataParam.csType);
```

Comment:
We should return null instead of empty array, to be consistent with structure properties code.